File: 2932. Maximum Strong Pair XOR I.js
// JAVASCRIPT
// 2932. Maximum Strong Pair XOR I
nums = [1,2,3,4,5]
/**
 * @param {number[]} nums
 * @return {number}
 */
//SOL
var maximumStrongPairXor = function(nums) {
    let max_xor = 0;
    const n = nums.length;
    
    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            let x = nums[i], y = nums[j];
            if (Math.abs(x - y) <= Math.min(x, y)) {
                max_xor = Math.max(max_xor, x ^ y);
            }
        }
    }
    
    return max_xor;
};
console.log(maximumStrongPairXor(nums))

       

File: 2932. Maximum Strong Pair XOR I.rs
// JAVASCRIPT
// 2932. Maximum Strong Pair XOR I
nums = [1,2,3,4,5]

//SOL
impl Solution {
    pub fn maximum_strong_pair_xor(nums: Vec<i32>) -> i32 {
        let mut max_xor = 0;
        let n = nums.len();
        
        for i in 0..n {
            for j in i..n {
                let x = nums[i];
                let y = nums[j];
                if (x - y).abs() <= x.min(y) {
                    max_xor = max_xor.max(x ^ y);
                }
            }
        }
        
        max_xor
    }
}

fn main() {
    println!("{}", Solution::maximum_strong_pair_xor(vec![1, 2, 3, 4, 5]));  // => 7
    println!("{}", Solution::maximum_strong_pair_xor(vec![10, 100]));        // => 0
    println!("{}", Solution::maximum_strong_pair_xor(vec![5, 6, 25, 30]));   // => 7
}


       